test/widgets/inbox_test.dart

```diff
@@ -310,6 +327,30 @@ void main() {
           checkAppearsUncollapsed(tester, findSectionContent);
         });
 
+        testWidgets('collapse all-DMs section after scroll', (tester) async {
+          await setupVarious(tester);
+
+           final listFinder = find.byType(Scrollable);
+          // Scroll the [StickyHeaderListView] enough so that
+          // the [_AllDmsSection] shows a sticky header
+          await tester.drag(listFinder, const Offset(0, -50));
+
+          final headerRow = findAllDmsHeaderRow(tester);
+          // Check that the sticky header is present
+          check(headerRow).isNotNull();
+
+          final findSectionContent = find.text(eg.otherUser.fullName);
+
+          checkAppearsUncollapsed(tester, findSectionContent);
+          await tapCollapseIcon(tester);
+          // Check that the header is still visible even after
+          // collapsing the section
+          check(headerRow).isNotNull();
+          checkAppearsCollapsed(tester, findSectionContent);
+          await tapCollapseIcon(tester);
+          checkAppearsUncollapsed(tester, findSectionContent);
```

Comment:
This test logic seems not to function as intended. As a result I'm not confident that this version would successfully accomplish the test's purpose, which is to fail if at some point in the future we make a change that accidentally reverts this feature.

Specifically, a test I like to do on any test for a new feature or bugfix is to run the test against the old code, from before the change. Here, that looks like this:
```
$ git checkout origin lib/
$ flutter test test/widgets/inbox_test.dart
00:03 +11: InboxPage collapsing all-DMs section collapse all-DMs section after scroll                 
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown running a test:
Finder specifies a widget that would not receive pointer events.
A call to tap() with finder "Found 1 widget with the given widget (Icon(IconData(U+0F102), size:
20.0, color: Color(0x7f1d2e48))): [
  Icon(IconData(U+0F102), size: 20.0, color: Color(0x7f1d2e48), dependencies: [Directionality,
IconTheme]),
]" derived an Offset (Offset(20.0, 46.0)) that would not hit test on the specified widget.
Maybe the widget is actually off-screen, or another widget is obscuring it, or the widget cannot
receive pointer events.
The finder corresponds to this RenderBox: RenderSemanticsAnnotations#e51b6 relayoutBoundary=up11(creator: Semantics ← Icon ← Padding ← Row ← Listener ← RawGestureDetector ← GestureDetector ← Semantics ← DefaultSelectionStyle ← Builder ← MouseRegion ← Semantics ← ⋯, parentData: offset=Offset(10.0, 10.0) (can use size), constraints: BoxConstraints(unconstrained), size: Size(20.0, 20.0))
The hit test result at that offset is: HitTestResult(_RenderInkFeatures#10262@Offset(20.0, 46.0),
RenderPhysicalModel#5f7e5@Offset(20.0, 46.0),
RenderAnnotatedRegion<SystemUiOverlayStyle>#7e1d9@Offset(20.0, 46.0),
RenderSemanticsAnnotations#f8204@Offset(20.0, 46.0), RenderConstrainedBox#d3925@Offset(20.0, 46.0),
RenderCustomMultiChildLayoutBox#c4b2b@Offset(20.0, 46.0), _RenderInkFeatures#7f830@Offset(20.0,
46.0), RenderPhysicalModel#a28d4@Offset(20.0, 46.0), RenderSemanticsAnnotations#1e887@Offset(20.0,
46.0), RenderRepaintBoundary#f1955@Offset(20.0, 46.0), RenderIgnorePointer#7d6b4@Offset(20.0, 46.0),
_RenderSnapshotWidget#8b907@Offset(20.0, 46.0), _RenderSnapshotWidget#ed2f3@Offset(20.0, 46.0),
_RenderSnapshotWidget#68834@Offset(20.0, 46.0), _RenderSnapshotWidget#0b54b@Offset(20.0, 46.0),
RenderRepaintBoundary#20409@Offset(20.0, 46.0), RenderSemanticsAnnotations#ae43e@Offset(20.0, 46.0),
RenderOffstage#141a0@Offset(20.0, 46.0), RenderSemanticsAnnotations#7000f@Offset(20.0, 46.0),
_RenderTheater#5ef0d@Offset(20.0, 46.0), RenderAbsorbPointer#a8ea2@Offset(20.0, 46.0),
RenderPointerListener#9d330@Offset(20.0, 46.0), RenderSemanticsAnnotations#a580a@Offset(20.0, 46.0),
RenderCustomPaint#e8d4b@Offset(20.0, 46.0), RenderSemanticsAnnotations#39568@Offset(20.0, 46.0),
RenderSemanticsAnnotations#8755e@Offset(20.0, 46.0), RenderTapRegionSurface#bfdb7@Offset(20.0,
46.0), RenderSemanticsAnnotations#a6989@Offset(20.0, 46.0),
RenderSemanticsAnnotations#3a436@Offset(20.0, 46.0), RenderSemanticsAnnotations#b7c7b@Offset(20.0,
46.0), HitTestEntry<HitTestTarget>#36c09(_ReusableRenderView#f920c),
HitTestEntry<HitTestTarget>#e0311(<AutomatedTestWidgetsFlutterBinding>))
If you expected this target not to be able to receive pointer events, pass "warnIfMissed: false" to
"tap()".
To make this error into a non-fatal warning, set WidgetController.hitTestWarningShouldBeFatal to
false.

When the exception was thrown, this was the stack:
#0      WidgetController._getElementPoint (package:flutter_test/src/controller.dart:1916:11)
#1      WidgetController.getCenter (package:flutter_test/src/controller.dart:1786:12)
#2      WidgetController.tap (package:flutter_test/src/controller.dart:1034:18)
#3      main.<anonymous closure>.<anonymous closure>.<anonymous closure>.tapCollapseIcon (file:///home/greg/z/flutterz/test/widgets/inbox_test.dart:277:24)
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///home/greg/z/flutterz/test/widgets/inbox_test.dart:350:17)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:183:15)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1018:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

The test description was:
  collapse all-DMs section after scroll
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +11 -1: InboxPage collapsing all-DMs section collapse all-DMs section after scroll [E]          
  Test failed. See exception logs above.
  The test description was: collapse all-DMs section after scroll
[…]
```

So:
* The test does fail on the old code. That's good!
* But it fails at line 350, in `tapCollapseIcon`. It does *not* fail at this check on line 348:

```dart
          // Check that the header is still visible even after
          // collapsing the section
          check(headerRow).isNotNull();
```

This test run was on the old code, from before your change, where #391 still exists and it is possible for the header to disappear when collapsing it. The test tried to check for that behavior, but it didn't find anything amiss when it did so. That means that either the test isn't successfully reproducing the behavior, or else the check it's making isn't successfully detecting the behavior.

So, please revise this test so that not only does it pass when paired with your changes, but it fails in a crisp way when run on the code from before your changes.